packages/common/src/common.ts

```diff
@@ -69,7 +71,7 @@ export class Common {
         (this._chainParams.customHardforks && this._chainParams.customHardforks[hf.name]),
     ])
     this._hardfork = this.DEFAULT_HARDFORK
-    this._params = { ...(opts.params ?? {}) } // copy
+    this._params = opts.params ? JSON.parse(JSON.stringify(opts.params)) : {} // copy
```

Comment:
Ok, so this is actually the single line change I initially intended, likely with the ones from `updateParams()` and `resetParams()` (good catch!), no changes intended for `copy()` whatsoever.

Will tests these lines in a new PR (maybe again, eventually you've been to this point already), then this should already be enough for this (or a replacing) PR!

(we also should not mix this with #3593 , it's absolutely not clear that these are related)